flake.nix

```diff
@@ -2,128 +2,107 @@
   description = "Seabug frontend";
 
   inputs = {
-    nixpkgs.url = github:NixOs/nixpkgs/nixos-unstable;
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-unstable;
     dream2nix = {
       url = github:nix-community/dream2nix;
       inputs.nixpkgs.follows = "nixpkgs";
     };
+    styleguide.url = github:mlabs-haskell/styleguide;
     # TODO: use `main` when this is merged
     seabug-contracts.url = github:mlabs-haskell/seabug-contracts/calum/fixing-buy-button;
   };
 
-  outputs =
-    { self
-    , nixpkgs
-    , dream2nix
-    , seabug-contracts
-    }:
+  outputs = inputs @  { self, nixpkgs, dream2nix, ... }:
     let
-      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+      systems = [ "x86_64-linux" "x86_64-darwin" ];
+      perSystem = nixpkgs.lib.genAttrs systems;
+      pkgs = perSystem (system: nixpkgs.legacyPackages.${system});
 
-      lib = nixpkgs.lib;
-
-      pkgsFor = lib.genAttrs supportedSystems (system: nixpkgs.legacyPackages.${system});
-
-      perSystem = lib.genAttrs supportedSystems;
-
-      d2nFlakeOutputs =
-        let
-          src = builtins.path {
-            path = ./.;
-            filter = path: _: builtins.baseNameOf path != "package-lock.json";
-          };
-        in
-        dream2nix.lib.makeFlakeOutputs {
-          systems = supportedSystems;
-          config.projectRoot = src;
-          source = src;
-          settings = [
-            {
-              subsystemInfo.noDev = false;
-              subsystemInfo.nodejs = 16;
-            }
-          ];
+      d2nFlake = dream2nix.lib.makeFlakeOutputs rec {
```

Comment:
Why using `rec` instead of `let`?
I'm always quite reluctant using `rec` when a `let` is sufficient, even if in this case `rec` would improve readability a bit I think.